tests/unit/multi-currency/test-class-currency-switcher-widget.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Class WCPay_Multi_Currency_Currency_Switcher_Widget_Tests
+ *
+ * @package WooCommerce\Payments\Tests
+ */
+
+use WCPay\Multi_Currency\Currency;
+
+/**
+ * WCPay\Multi_Currency\Currency_Switcher_Widget unit tests.
+ */
+class WCPay_Multi_Currency_Currency_Switcher_Widget_Tests extends WP_UnitTestCase {
+	/**
+	 * Mock WCPay\Multi_Currency\Multi_Currency.
+	 *
+	 * @var WCPay\Multi_Currency\Multi_Currency|PHPUnit_Framework_MockObject_MockObject
+	 */
+	private $mock_multi_currency;
+
+	/**
+	 * WCPay\Multi_Currency\Currency_Switcher_Widget instance.
+	 *
+	 * @var WCPay\Multi_Currency\Currency_Switcher_Widget
+	 */
+	private $currency_switcher_widget;
+
+	/**
+	 * Pre-test setup
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->mock_multi_currency = $this->createMock( WCPay\Multi_Currency\Multi_Currency::class );
+		$this->mock_multi_currency
+			->method( 'get_enabled_currencies' )
+			->willReturn(
+				[
+					new Currency( 'USD' ),
+					new Currency( 'CAD', 1.2 ),
+					new Currency( 'EUR', 0.8 ),
+				]
+			);
+
+		$this->currency_switcher_widget = new WCPay\Multi_Currency\Currency_Switcher_Widget( $this->mock_multi_currency );
+	}
+
+	public function test_widget_output() {
```

Comment:
I feel like this test can be split to avoid testing many things at once without having too much of a performance penalty.

What do you think of splitting it into?
- `test_widget_renders_title_with_args`
- `test_widget_renders_enabled_currencies_with_symbol`
- `test_widget_renders_enabled_currencies_without_symbol`
- `test_widget_renders_enabled_currencies_with_symbol_and_flag`
- `test_widget_selects_selected_currency`
- `test_widget_submits_form_on_change`